app.py

```diff
@@ -0,0 +1,148 @@
+
+app = Flask(__name__)
+
+app.config["SECRET_KEY"] = os.environ.get('SECERT_KEY', SECRET_KEY)
+app.config["SQLALCHEMY_DATABASE_URI"] = os.environ.get(
+    'DATABASE_URL', "postgresql://localhost/chargR?user=postgres&password=postgresql")
+app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
+app.config['SQLALCHEMY_ECHO'] = True
+app.config['DEBUG_TB_INTERCEPT_REDIRECTS'] = False
+
+
+API_BASE_URL = 'https://api.openchargemap.io/v3/poi/'
+MAP_BASE_URL = 'http://open.mapquestapi.com/geocoding/v1/'
+
+# toolbar = DebugToolbarExtension(app)
+
+connect_db(app)
+db.create_all()
+
+
+@app.route('/')
+def start():
+    return redirect('/home')
+
+
+@app.route('/home')
+def home_page():
+    return render_template('home.html')
+
+########################### Location And Coords ######################################
+
+
+def get_coords(location):
+    """Get the location and return Latitude and Longitude """
+
+    res = requests.get(f'{MAP_BASE_URL}/address',
+                       params={'key': MAP_KEY, 'location': location})
+    data = res.json()
+    lat = data['results'][0]['locations'][0]['latLng']['lat']
+    lng = data['results'][0]['locations'][0]['latLng']['lng']
+    coords = {'lat': lat, 'lng': lng}
+    return coords
+
+
+def get_results(data):
+    """turn json into python"""
+
+    res = json.loads(data.text)
+    return res
+
+
+def get_info(coords):
+    """Get the coords and return data """
+    latitude = coords['lat']
+    longitude = coords['lng']
+
+    response = requests.get(f'{API_BASE_URL}', params={'key': OPEN_CHARGE_MAP_KEY,
+                            'countrycode': 'US', 'latitude': latitude, 'longitude': longitude})
+
+    return get_results(response)
+
+
+@ app.route('/results')
+def search_result():
+    """"Show the all results """
+    location = request.args['location']
+
+    coords = get_coords(location)
+    result = get_info(coords)
+
+    return render_template('result.html', result=result)
+
+######################## Login/Signup/Logout #######################
+
+
+@ app.route('/login', methods=['GET', 'POST'])
+def login_page():
+    """" User Login"""
+    form = LoginForm()
+
+    if form.validate_on_submit():
+        username = form.username.data
+        password = form.password.data
+
+        user = User.authenticate(username, password)
+        if user:
+            session['username'] = user.username
+            flash(f'Welcome back!,{user.username}', 'success')
+            return redirect(f'/home')
+        else:
+            form.username.errors = ['Invalid usename/password']
+
+    return render_template('login.html', form=form)
+
+
+@app.route('/logout')
+def logout():
+    """" User Logout """
+    session.pop('username')
+    flash('Successfully logout', 'success')
+    return redirect('/home')
+
+
+@ app.route('/signup', methods=['GET', 'POST'])
+def signup_page():
+    """" User SignUp"""
+    form = SignUpForm()
+
+    if form.validate_on_submit():
+        username = form.username.data
```

Comment:
Not mandatory, but I suggest removing the `username` field and using the email as the unique username.